python/ooi_data_explorations/uncabled/process_adcp.py

```diff
@@ -0,0 +1,307 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+QCThresholds = {
+    'error_velocity': {
+        'pass': 0.05,
+        'fail': 0.20 },
+    'correlation_magnitude': {
+        'pass': 115,
+        'fail': 63 },
+    'vertical_velocity': {
+        'pass': 0.30,
+        'fail': 0.50 },
+    'horizontal_speed': {
+        'pass': 1.00,
+        'fail': 2.50 },
+    'percent_good': {
+        'ADCPT': {
+            'pass': 56,
+            'fail': 45 },
+        'ADCPS': {
+            'pass': 48,
+            'fail': 38 }
+    }
+}
+
+
+def sidelobe_depth(ds, theta=20):
+    """
+    Calculate the sidelobe contamination depth for the given ADCP.
+
+    The sidelobe intereference depth 
+    is caluclated following Lentz et al. (2022) where:
+    
+        z_ic = ha*[1 - cos(theta)] + 3*delta_Z/2
+        
+    z_ic is the depth above which there is sidelobe interference,
+    ha is the transducer face depth, theta is the beam angle, and
+    delta_Z is the cell-bin depth. At this time we ignore instrument 
+    tilt at this time and its impact, assuming fixed beam angle.
```

Comment:
"At this time we ignore instrument tilt at this time" - grammar QC check. Very important.